### PR TITLE
dev(replays): add profiling to replay recording consumer

### PR DESCRIPTION
**Pull request by JoshFerge**

### CODE REVIEW

File: src/sentry/replays/consumers/recording/process_recording.py [hunk 1]
```diff
@@ -74,15 +75,18 @@ def _process_chunk(
...
         current_transaction: Transaction,
     ) -> None:
-        cache_prefix = replay_recording_segment_cache_id(
-            project_id=message_dict["project_id"],
-            replay_id=message_dict["replay_id"],
-            segment_id=message_dict["id"],
-        )
+        with current_transaction.start_child(
+            op="replays.process_recording._process_chunk", description="process_chunk"
+        ), start_profiling(current_transaction, current_transaction.hub):
```
Comment (Zylphrex) [inline src/sentry/replays/consumers/recording/process_recording.py:80]: I haven't tested how this call will be affected if there's other work between the start of the transaction and the start of the profile. I believe what will happen is that the start/end of the profile will no longer align with the transaction so the result might be slightly offset.

**Review by Zylphrex — APPROVED** [review]
Note this alone won't start sending profiles. You'll have to add the environment variable to your consumer. And do keep an eye on the CPU usage when this is deployed. I expect it to grow, but it should be within reason.

**Comment by JoshFerge** [comment]
> Note this alone won't start sending profiles. You'll have to add the environment variable to your consumer. And do keep an eye on the CPU usage when this is deployed. I expect it to grow, but it should be within reason.

we have our sampling rate set extremely low (.02%), so should be okay.

**Comment by Zylphrex** [comment]
> > Note this alone won't start sending profiles. You'll have to add the environment variable to your consumer. And do keep an eye on the CPU usage when this is deployed. I expect it to grow, but it should be within reason.
> 
> we have our sampling rate set extremely low (.02%), so should be okay.

One thing to keep in mind is that the sampling rate largely only affects how much data is sent out. Since the way the profiler works by running a background thread to capture samples, as long as there is a sampled transaction active, it'll be capturing samples.

**Pull request closed without merging**